v3/src/components/graph/adornments/univariate-measures/univariate-measure-adornment-component.tsx

```diff
@@ -87,119 +120,191 @@ export const UnivariateMeasureAdornmentComponent = observer(
       const labelHeight = labelNode?.getBoundingClientRect().height || 0
       const x = event.x - labelWidth / 2
       const y = event.y - labelHeight / 2
-
       const measure = measures.get(instanceKey)
       measure?.setLabelCoords({x, y})
     }, [instanceKey, model])
 
-    const addLineCoverAndLabel = useCallback((
-      valueObj: IValueObject, labelObj: ILabelObject, measure: IMeasureInstance
+    const newLine = useCallback((lineClass: string, lineId: string, coords: ILineCoords) => {
+      const selection = select(valueRef.current)
+      const { x1, x2, y1, y2 } = coords
+      return selection.append("line")
+        .attr("class", lineClass)
+        .attr("id", lineId)
+        .attr("data-testid", lineId)
+        .attr("x1", x1)
+        .attr("x2", x2)
+        .attr("y1", y1)
+        .attr("y2", y2)
+    }, [])
+
+    const addRange = useCallback((
+      range: number, coverClass: string, lineClass: string, valueObj: IValue, coords: ILineCoords
+    ) => {
+      if (!dataConfig || (!isStandardDeviationAdornment(model) && !isMeanAbsoluteDeviationAdornment(model))) return
```

Comment:
How about adding a property like `hasRange` to the registration or having a view on the model that returns true for `hasRange` so that we don't have to hard-code specific types here?

Reply:
@kswenson I tried exactly that first, but there was a problem. I think for this particular instance, we can actually drop the adornment type check since it's also done in `addLineCoverAndLabel` before that function calls `addRange`. Still, if I add a `hasRange` property to the `UnivariateMeasureAdornmentModel`, set it to false by default, and set it to true on the SD and MAD adornment models, the code below will return the error `Property 'computeMeasureRange' does not exist on type 'IUnivariateMeasureAdornmentModel'.`

```
const measureRange = attrId && model.hasRange
  ? model.computeMeasureRange(attrId, cellKey, dataConfig)
  : undefined
```
I couldn't figure out another way to make it work other than to use `isStandardDeviationAdornment` and `isMeanAbsoluteDeviationAdornment` instead. Are you aware of a straightforward way to make the `hasRange` check work the way we'd want it to? Maybe adding a default `computeMeasure` view to `UnivariateMeasureAdornmentModel` that returns `NaN` or `undefined` is a reasonable option?

Reply:
Add a `hasRange` view to `UnivariateMeasureAdornmentModel` that returns `false`.